server/src/modules/user/user.controller.ts

```diff
@@ -112,4 +124,46 @@ export class UserController {
       throw new BadRequestException(error.response);
     }
   }
+
+  @PublicRoute()
+  @Post('/admin')
+  async createNewAdminUser(
+    @Headers() headers: HeaderDto,
+    @Body() body: SignUpRequestBodyDto,
+    @Res({ passthrough: true }) res: Response,
+  ): Promise<IUserResponse> {
+    try {
+      const xApiKey = headers['x-api-key'];
+
+      if (!xApiKey)
+        throw new UnauthorizedException('x-api-key key not provided');
+
+      // call User Service to varify and resister admin user
+      const { _id, role } = await this.userService.registerAdminUser(
+        xApiKey,
+        body,
+      );
+
+      const { token, refreshToken } = await this.userService.generateTokens(
+        _id,
+        role,
+      );
```

Comment:
call the single service file's method and do all the processes into service file

Reply:
updated also for signup and login